data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -17,17 +17,25 @@ OPENSHIFT_HYPERKUBE_IMAGE=$(podman run --rm {{.ReleaseImage}} image hyperkube)
 
 mkdir --parents ./{bootstrap-manifests,manifests}
 
+# convert the release image pull spec to an "absolute" form if a digest is available - this is
+# safe to resolve after the actions above because podman will not pull the image once it is 
+# locally available
+if ! release=$( podman inspect {{.ReleaseImage}} -f '{{"{{"}} index .RepoDigests 0 {{"}}"}}' ) || [[ -z "${release}" ]]; then
+	echo "Warning: Could not resolve release image to pull by digest" 2>&1
+	release="{{.ReleaseImage}}"
```

Comment:
Is this case a thing?  How can you have an image that you could successfully pull but which doesn't have a digest?

Reply:
if someone provides a registry that isn't perfectly compatible with v2_2.

Reply:
> if someone provides a registry that isn't perfectly compatible with v2_2.

Then we can't digest the image?  How do references work in pre-`v2_2`?  It's still a Merkle DAG, right?  If we're relying on *registries* to tell us what digests are, we have big problems.  And aren't all of the release-payload operator references based on digests?